internal/api/v1alpha1/pod_spec_mutation_config.go

```diff
@@ -73,6 +75,10 @@ type PodTemplateSpecMutationConfig struct {
 	// +kubebuilder:default:=false
 	PurgeAnnotations bool `json:"purgeAnnotations,omitempty"`
 
+	// PatchSpecOperations contains a list of JSON patch operations to apply to the PodSpec.
+	// [`JSONPatch`](https://www.rfc-editor.org/rfc/rfc6902.html)
+	PatchSpecOperations []map[string]string `json:"patchSpecOperations,omitempty"`
```

Comment:
Hmm. Is there not a better type we can use here? I could put in an easily mis-typed or incorrect patch:

```
    patchSpecOperations:
      - notAnOp: replace
        mutatePath: '/spec/containers/0/name'
        badValue: dennis
```

Reply:
I'll see if we can do sort of validation or a better type here. The current third party library used in this PR uses just `[]bytes`.

Reply:
Perhaps we can build our own JSNOPatch type - and also use https://github.com/evanphx/json-patch/blob/master/v5/patch.go#L905-L918

Reply:
Yea i tried to create own JSON Patch type but looks using `interface{}` for `value` causes issues from `controller-tools` [ref](https://github.com/kubernetes-sigs/controller-tools/pull/126). We could update to `string` if anything? 

```
type JSONPatchOperation struct {
	// +kubebuilder:validation:Required
	Operation string `json:"op"`
	// +kubebuilder:validation:Required
	Path string `json:"path"`
	// +kubebuilder:validation:Optional
	Value interface{} `json:"value,omitempty"`
}
```

Reply:
Ultimately went with 
```
// JSONPatchOperation represents a JSON Patch operation defined in https://www.rfc-editor.org/rfc/rfc6902.html
type JSONPatchOperation struct {
	// +kubebuilder:validation:Required
	Operation string `json:"op"`
	// +kubebuilder:validation:Required
	Path string `json:"path"`
	// +kubebuilder:validation:Optional
	Value intstr.IntOrString `json:"value,omitempty"`
}
```

manually verified basic validations as well...

``` 
➜  examples git:(add-json-patches-feat) ✗ k apply -f pod_access_template.yaml

The PodAccessTemplate "deployment-example" is invalid: spec.controllerTargetMutationConfig.patchSpecOperations[0].op: Invalid value: "integer": spec.controllerTargetMutationConfig.patchSpecOperations[0].op in body must be of type string: "integer"
```

validatePatch gets called explicitly by `DecodePatch` which we already call as well.